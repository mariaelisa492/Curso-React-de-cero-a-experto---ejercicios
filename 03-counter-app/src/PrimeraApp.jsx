import React from "react";   //tengo que importar react porque estoy usando jsx
import PropTypes from 'prop-types';


export const PrimeraApp = ({saludo, anio})=>{

    // const saludo = 'Hola Mundo';
    // const anio = 2021;     
    
    return (                                                  //debo poner todo dentro de un solo div que los contenga o en el fragment <></>
        <>
        {/* {<pre>{JSON.stringify(saludo, null, 3)}</pre>} */}
        <h1>{saludo}!!!</h1>
        <p>{anio}</p>
        </>
    )
}

PrimeraApp.protoTypes = {
    saludo: PropTypes.string.isRequired,
}

PrimeraApp.defaultProps = {                  //para poner props por default
    anio: '2021'
}



// export default  PrimeraApp;







//vamos a trabajar con FUNCTIONAL COMPONENTES no con componentes de clase
//Este es un functional component exportado