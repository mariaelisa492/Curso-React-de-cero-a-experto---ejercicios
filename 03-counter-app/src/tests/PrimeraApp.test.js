// import { render } from "@testing-library/react";
import React from 'react';
import '@testing-library/jest-dom'
import{shallow} from 'enzyme'
import { PrimeraApp } from "../PrimeraApp";


describe('Pruebas en <PrimeraApp />', () => {
    

    // test('Debe mostrar el mensaje "Hola, soy Goku"', () => {
    //     // const saludo = 'Hola, soy Gokusiiii';
    //     // const {getByText} = render(<PrimeraApp saludo={saludo} />)
    //     // expect(getByText(saludo)).toBeInTheDocument();  //deberias esperar que exista el saludo en el documento
    // })

    test('Debe mostrar <PrimeraApp /> correctamente', () => {

        const saludo = 'Hola, soy Goku';

        const wrapper = shallow(<PrimeraApp saludo={saludo} />)   

        expect(wrapper).toMatchSnapshot();    

    })

    test('Debe mostrar el aÃ±o mostrado enviado en props ', () => {

        const saludo = 'Hola, soy Goku';
        const anio = '2021';

        const wrapper = shallow(
                <PrimeraApp 
                saludo = {saludo}
                anio = {anio}
                />
            );

        const textoParrafo = wrapper.find('p').text();

        expect(textoParrafo).toBe(anio)

        
    })
    
    


    
})
