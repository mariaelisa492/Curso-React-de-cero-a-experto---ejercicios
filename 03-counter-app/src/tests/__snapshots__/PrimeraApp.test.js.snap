// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <PrimeraApp /> Debe mostrar <PrimeraApp /> correctamente 1`] = `
<Fragment>
  <h1>
    Hola, soy Goku
    !!!
  </h1>
  <p>
    2021
  </p>
</Fragment>
`;
